nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/executor/WorkflowStateProcessor.java

```diff
@@ -239,63 +251,155 @@ private WorkflowActionType getActionType(StateExecutionImpl execution) {
   }
 
   private boolean isNextActivationImmediately(StateExecutionImpl execution) {
-    return execution.getNextActivation() != null && !execution.getNextActivation().isAfterNow();
+    return execution.isStateProcessed() && execution.getNextActivation() != null && !execution.getNextActivation().isAfterNow();
```

Comment:
why we are not processing this instance immediately when the state was not processed? depending on what the filter returned, it may be the next one in the queue anyway